### ISSUE TITLE
How to identify/visualize co-expression (correlation) between markers?

### ISSUE THREAD

**Issue by WilliamMWei** [ImmuneDynamics/Spectre#182]
Hi,
Thank you so much for the tool.

Could you give me some pointers on packages/functions to identify/visualize the co-expression (correlations) between markers or of one marker with every other markers?

Thank you so much for your help.

---
I have also posted in Catalyst github for query.
https://github.com/HelenaLC/CATALYST/issues/379

**Comment by tomashhurst** [comment]
Hi @WilliamMWei , do you mean something a bit like this (images attached)? In this case we are correlating various features, but this can be used in a similar way to look at the correlation of marker expresison.

![image](https://github.com/ImmuneDynamics/Spectre/assets/11766139/8595520b-3e7b-47ad-9524-0ebd6a745464)
![image](https://github.com/ImmuneDynamics/Spectre/assets/11766139/9eeb90a9-9c18-46c6-aba1-493fcd3d983a)

From this paper: https://www.sciencedirect.com/science/article/pii/S2666379121000197?via%3Dihub#fig6

**Comment by SamGG** [comment]
For visualising data frames with mixed data types, look at [FunkyHeatmap](https://funkyheatmap.github.io/funkyheatmap/).

**Comment by denvercal1234GitHub** [comment]
Hi @tomashhurst - I wonder if you guys are planning to release updated tutorial version soon? If so, it would be awesome if we could have this exploration of correlation heatmap given the availability of already useful bits at https://immunedynamics.io/spectre/cytometry/#tutorials.

**Comment by ghar1821** [comment]
I don't think we have any functions that can do correlation/co-expression analysis. We will need to look into this.

In the meantime, I just realised that I actually have implemented a function that can create the plot below some time ago (https://github.com/ImmuneDynamics/Spectre/blob/development/R/make.bubble.plot.R). I think I used it once for something but somehow never used it again. The plot can show correlation between different markers for each cell type. Maybe give it a try and let me know if it works or not?

> ![image](https://private-user-images.githubusercontent.com/11766139/292058509-8595520b-3e7b-47ad-9524-0ebd6a745464.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwMDczNDQsIm5iZiI6MTcyMTAwNzA0NCwicGF0aCI6Ii8xMTc2NjEzOS8yOTIwNTg1MDktODU5NTUyMGItM2U3Yi00N2FkLTk1MjQtMGViZDZhNzQ1NDY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE1VDAxMzA0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxY2VlNzJkMTg2YmEwNDNmMzg0OTZhZGM5MWQzMjg4ZjNjZDg1ZWI5MTEzYjBiZjY3N2RmY2M2MDdiMTA1MGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.FZxD0MlHsMMrVKipDHyOHf8HdcIKaV0wO1n55hxWRAc) ![image](https://private-user-images.githubusercontent.com/11766139/292058581-9eeb90a9-9c18-46c6-aba1-493fcd3d983a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwMDczNDQsIm5iZiI6MTcyMTAwNzA0NCwicGF0aCI6Ii8xMTc2NjEzOS8yOTIwNTg1ODEtOWVlYjkwYTktOWMxOC00NmM2LWFiYTEtNDkzZmNkM2Q5ODNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE1VDAxMzA0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY1ZjZiZTQ2YTE2YWIxNjNlNzM1OWU1MjU3MGQ4ZDdlYThhYzkzYTJmZWJiYzRmZDVkMDdjYmM3MWI2NDllNzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.K2Lzw6DrQ52Ghg_fwVOFYtH5X6y-ckOoiPQGaaJ7kjY)
